tests/core/go_path/pkg/lib/generated_embeded.bzl

```diff
@@ -0,0 +1,107 @@
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load(
+    "@io_bazel_rules_go//go:def.bzl",
+    "go_context",
+    "go_library",
+)
+
+def _gen_library_impl(ctx):
+    go = go_context(ctx)
+    libname = getattr(ctx.attr, "libname")
+    src = go.actions.declare_file(ctx.label.name + ".go")
+
+    embedsrcs = getattr(ctx.attr, "embedsrcs", [])
+
+    embed_declarations = ""
+    i = 0
+    for e in embedsrcs:
+        for f in e.files.to_list():
+            embed_declarations += "//go:embed {0}\nvar embeddedSource{1} string\n".format(f.basename, i)
+            i += 1
+
+    go.actions.write(src, "package " + libname + "\n" +
```

Comment:
Can you use [expand_template](https://bazel.build/rules/lib/actions#expand_template) instead? It's much harder to read with all the escapes

Reply:
You'll still have the same issues with `expand_template` I mage some changes to improve the readability